plugins/providers/hyperv/config.rb

```diff
@@ -24,21 +55,20 @@ def initialize
         @vmname = UNSET_VALUE
         @vlan_id = UNSET_VALUE
         @mac = UNSET_VALUE
+        @linked_clone = UNSET_VALUE
         @differencing_disk = UNSET_VALUE
         @auto_start_action = UNSET_VALUE
         @auto_stop_action = UNSET_VALUE
         @enable_virtualization_extensions = UNSET_VALUE
-        @vm_integration_services = {
-            guest_service_interface: UNSET_VALUE,
-            heartbeat: UNSET_VALUE,
-            key_value_pair_exchange: UNSET_VALUE,
-            shutdown: UNSET_VALUE,
-            time_synchronization: UNSET_VALUE,
-            vss: UNSET_VALUE
-        }
+        @enable_checkpoints = UNSET_VALUE
+        @vm_integration_services = {}
       end
 
       def finalize!
+        @linked_clone = false if @linked_clone == UNSET_VALUE
+        @differencing_disk = false if @differencing_disk == UNSET_VALUE
```

Comment:
Wouldn't some sort of deprecation warning make sense somewhere around here? If it is deprecated, perhaps telling the user makes sense.